src/refdb_fs.c

```diff
@@ -962,6 +962,8 @@ static int packed_write(refdb_fs_backend *backend)
 
 	for (i = 0; i < git_sortedcache_entrycount(refcache); ++i) {
 		struct packref *ref = git_sortedcache_entry(refcache, i);
+		if (!ref)
+			goto fail;
```

Comment:
Can this happen? We've taken a write lock and are iterating over the amount of entries which the data structure has told us exist. If we're shut up a warning, `assert(ref)` would be more appropriate imo.